apps/aecore/lib/aecore/channel/tx/channel_snapshot_solo.ex

```diff
@@ -0,0 +1,205 @@
+  def validate(%ChannelSnapshotSoloTx{state: %ChannelStateOffChain{sequence: sequence}}, data_tx) do
+    senders = DataTx.senders(data_tx)
+
+    cond do
+      length(senders) != 1 ->
+        {:error, "#{__MODULE__}: Invalid senders size"}
+
+      sequence == 0 ->
+        {:error, "#{__MODULE__}: Can't slash with zero state"}
+
+      true ->
+        :ok
+    end
+  end
+
+  @doc """
+  Snapshots a channel.
+  """
+  @spec process_chainstate(
+          Chainstate.account(),
+          ChannelStateTree.t(),
+          non_neg_integer(),
+          ChannelSnapshotSoloTx.t(),
+          DataTx.t()
+        ) :: {:ok, {Chainstate.accounts(), ChannelStateTree.t()}}
+  def process_chainstate(
+        accounts,
+        channels,
+        _block_height,
+        %ChannelSnapshotSoloTx{
+          state:
+            %ChannelStateOffChain{
+              channel_id: channel_id
+            } = state
+        },
+        _data_tx
+      ) do
+    new_channels =
+      ChannelStateTree.update!(channels, channel_id, fn channel ->
+        ChannelStateOnChain.apply_snapshot(channel, state)
+      end)
+
+    {:ok, {accounts, new_channels}}
+  end
+
+  @doc """
+  Checks whether all the data is valid according to the ChannelSnapshotTx requirements,
+  before the transaction is executed.
+  """
+  @spec preprocess_check(
+          Chainstate.account(),
+          ChannelStateTree.t(),
+          non_neg_integer(),
+          ChannelSnapshotSoloTx.t(),
+          DataTx.t()
+        ) :: :ok | {:error, String.t()}
+  def preprocess_check(
+        accounts,
+        channels,
+        _block_height,
+        %ChannelSnapshotSoloTx{state: state},
+        data_tx
+      ) do
+    sender = DataTx.main_sender(data_tx)
+    fee = DataTx.fee(data_tx)
+
+    channel = ChannelStateTree.get(channels, state.channel_id)
+
+    cond do
+      AccountStateTree.get(accounts, sender).balance - fee < 0 ->
+        {:error, "#{__MODULE__}: Negative sender balance"}
+
+      channel == :none ->
+        {:error, "#{__MODULE__}: Channel doesn't exist (already closed?)"}
+
+      !ChannelStateOnChain.active?(channel) ->
+        {:error, "#{__MODULE__}: Can't submit snapshot when channel is closing (use slash)"}
+
+      true ->
+        ChannelStateOnChain.validate_snapshot(channel, state)
+   end
+  end
+
+  @spec deduct_fee(
+          Chainstate.accounts(),
+          non_neg_integer(),
+          ChannelSnapshotSoloTx.t(),
+          DataTx.t(),
+          non_neg_integer()
+        ) :: Chainstate.account()
+  def deduct_fee(accounts, block_height, _tx, data_tx, fee) do
+    DataTx.standard_deduct_fee(accounts, block_height, data_tx, fee)
+  end
+
+  @spec is_minimum_fee_met?(SignedTx.t()) :: boolean()
+  def is_minimum_fee_met?(tx) do
+    tx.data.fee >= Application.get_env(:aecore, :tx_data)[:minimum_fee]
+  end
+
+  def encode_to_list(%ChannelSnapshotSoloTx{} = tx, %DataTx{} = datatx) do
```

Comment:
It's datatx here and data_tx above, please make it consistent, also some structure fields are matched and some are accessed with ".",  could you change that to be consistent as well?